api/tests/test_public_canteen_previews.py

```diff
@@ -113,7 +113,9 @@ def test_recieve_expected_keys_in_preview(self):
         Should get summary data in a public preview
         """
         canteen = CanteenFactory.create(publication_status=Canteen.PublicationStatus.PUBLISHED)
-        DiagnosticFactory.create(canteen=canteen)
+        DiagnosticFactory.create(
+            canteen=canteen, year=2020
```

Comment:
une alternative non-hardodée serait `timezone.now().date().year - 1`
(en regardant le code dans Canteen > appro_diagnostics)

Reply:
en générale j’évite d’écrire de la logique dans les tests quand c'est possible d'écrire sans (pour le code normale bien sûr éviter tout hard-codage). Est-ce que tu vois des pbs potentiels avec le fait de hardcoder ça dans un test ?

Reply:
ok up touyou :)

un peu plus haut dans le code il y a `date.today().year - 1`. et il y a aussi une variété de 2019, 2020 et 2021.

Reply:
ca pourrait limite être une valeur par défaut dans le DiagnosticFactory (avec un commentaire "in the past")

Reply:
ah oui j'ai pas vu que c'est déjà écrit. Bref, pourquoi pas